app/models/pulitzer/post.rb

```diff
@@ -1,11 +1,12 @@
 module Pulitzer
   class Post < ActiveRecord::Base
     extend ::FriendlyId
+    has_many :versions
     belongs_to :post_type
-    has_many :content_elements, dependent: :destroy
-    has_many :post_tags, dependent: :destroy
     delegate :post_type_content_element_types, to: :post_type
+    delegate :content_elements, :post_tags, to: :preview_version
```

Comment:
In the reservation -> option setup, we made it so that Reservation.balance or Reservation.units always defaulted to the active option. This way, when anything important is done with a reservation, we know it will always get the right data.

Then, for preview options we explicitly operate on the option itself, not the Reservation. The result is that the reservation can just delegate to the active option.

This would make it simpler when implementing a public page if you could just to Post.content_elements and know it would always be the active ones.

Does this make sense? It works either way, it's just a little cleaner when you're actually calling up data for it's final purpose if you don't have to think about versions.

When you're editing a Post, you can just do `preview_version = Post.preview_version`, and then do all the editing on `preview_version`

Let's chat on this if you have any questions.